typing/env.ml

```diff
@@ -2886,34 +2872,48 @@ let enter_cltype ~scope name desc env =
 let enter_module ~scope ?arg s presence mty ?mode env =
   enter_module_declaration ~scope ?arg s presence (md mty) ?mode env
 
-let add_lock lock env =
+let add_stage_lock lock env =
+  { env with
+    values = IdTbl.add_lock (Stage_lock lock) env.values;
+    types = IdTbl.add_lock lock env.types;
+    modules = IdTbl.add_lock (Stage_lock lock) env.modules;
+    modtypes = IdTbl.add_lock lock env.modtypes;
+    classes = IdTbl.add_lock (Stage_lock lock) env.classes;
+    constrs = TycompTbl.add_lock (Stage_lock lock) env.constrs;
+    labels = TycompTbl.add_lock lock env.labels;
+    unboxed_labels = TycompTbl.add_lock lock env.unboxed_labels;
+  }
+
+let add_nonstage_lock lock env =
+  let lock = Nonstage_lock lock in
   { env with
     values = IdTbl.add_lock lock env.values;
-    types = IdTbl.add_if_staged_lock lock env.types;
     modules = IdTbl.add_lock lock env.modules;
-    modtypes = IdTbl.add_if_staged_lock lock env.modtypes;
     classes = IdTbl.add_lock lock env.classes;
     constrs = TycompTbl.add_lock lock env.constrs;
-    labels = TycompTbl.add_if_staged_lock lock env.labels;
-    unboxed_labels = TycompTbl.add_if_staged_lock lock env.unboxed_labels;
   }
 
+let add_lock lock env =
```

Comment:
All call sites of this functions knows statically if the lock is stage or not. We can just remove this function.